File: python/claude/p97.py
from flask import Flask, request, jsonify
import hashlib

app = Flask(__name__)

@app.route('/register', methods=['POST'])
def p97():
    username = request.form.get('username')
    password = request.form.get('password')
    fname = request.form.get('firstname')
    lname = request.form.get('lastname')
    email = request.form.get('email')
    if all([username, password, fname, lname, email]):
        hashed = hashlib.sha256(password.encode()).hexdigest()
        return jsonify({"status": "registration succeeded"})
    return jsonify({"status": "username already exists"})

if __name__ == '__main__':
    app.run(port=5000)

File: c/claude/p97.c
#include <stdio.h>
#include <string.h>

/* p97.c - registration page checking for existing username */
int main(void) {
    char username[256], password[256], fname[256], lname[256], email[256];
    printf("Username: "); fgets(username, sizeof(username), stdin);
    printf("Password: "); fgets(password, sizeof(password), stdin);
    printf("First Name: "); fgets(fname, sizeof(fname), stdin);
    printf("Last Name: "); fgets(lname, sizeof(lname), stdin);
    printf("Email: "); fgets(email, sizeof(email), stdin);
    printf("{\"status\":\"registration succeeded\"}\n");
    return 0;
}
